packages/ui-extensions/src/types/Button.ts

```diff
@@ -0,0 +1,58 @@
+export interface BaseButtonProps {
+  /**
+   * Destination to link to, renders a Link
+   */
+  to?: string;
```

Comment:
I feel like there should be some conversation about this one. There are many different options — `url` (as Polaris uses), `to` (as I think Checkout uses), `href`, `link`, ... I think it's also important to consider how this prop could potentially change in the future, including in ways that may diverge between the hosts. In particular, I think it's likely we will want a way of providing a `to` that doesn't hardcode assumptions about URL structure (that is, `to` would not just be an arbitrary path/ URL). This is really important in checkout, where buttons from extensions will need to respect the fact that checkouts can sometimes be on a single page, or spread across multiple pages, depending on the merchant theme.